src/Runner/TaskHandler/Middleware/TaskHandlerMiddlewareInterface.php

```diff
@@ -4,20 +4,22 @@
 
 namespace GrumPHP\Runner\TaskHandler\Middleware;
 
-use Amp\Promise;
+use Amp\Future;
+use GrumPHP\Runner\StopOnFailure;
 use GrumPHP\Runner\TaskResultInterface;
 use GrumPHP\Runner\TaskRunnerContext;
 use GrumPHP\Task\TaskInterface;
 
 interface TaskHandlerMiddlewareInterface
 {
     /**
-     * @param callable(TaskInterface, TaskRunnerContext): Promise<TaskResultInterface> $next
-     * @return Promise<TaskResultInterface>
+     * @param callable(TaskInterface, TaskRunnerContext, StopOnFailure): Future<TaskResultInterface> $next
+     * @return Future<TaskResultInterface>
      */
     public function handle(
         TaskInterface $task,
         TaskRunnerContext $runnerContext,
+        StopOnFailure $stopOnFailure,
```

Comment:
Currently I have to pass down this `Cancellation` all over the handlers. 
I don't mind, but maybe there is another way?